modules/client/src/main/java/org/apache/ignite/internal/client/sql/ClientSql.java

```diff
@@ -287,6 +291,38 @@ public <T> CompletableFuture<AsyncResultSet<T>> executeAsync(
             @Nullable CancellationToken cancellationToken,
             Statement statement,
             @Nullable Object... arguments) {
+        return executeAsyncInternal(
```

Comment:
let's add a few test cases to `org.apache.ignite.client.ClientSqlTest` to make sure serialization works as expected